ast/compile.go

```diff
@@ -347,6 +385,22 @@ func (c *Compiler) GetRules(ref Ref) (rules []*Rule) {
 	return rules
 }
 
+// GetFunc returns the function referred to by name.
+func (c *Compiler) GetFunc(name String) []*Func {
+	if fn, ok := c.FunctionMap[name]; ok {
+		return fn
+	}
+	return nil
+}
+
+// GetAllFuncs returns a list of functions that this compiler has discovered.
+func (c *Compiler) GetAllFuncs() (funcs [][]*Func) {
+	for _, fn := range c.FunctionMap {
```

Comment:
No explicit need for the list, but I wanted to return a copy so someone couldn't mess with the compiler's functions from the outside.